### PR TITLE
Decompile audioMgr

### PR DESCRIPTION
**Pull request by Thar0**
Decompiles audioMgr.c
One non-matching remaining

Guidance on how to refactor includes to avoid putting the AudioMgr struct in its own header are very welcome.

### CODE REVIEW

File: src/code/audioMgr.c [hunk 1]
```diff
@@ -1,16 +1,130 @@
...
+// ? func_800C3CB8(?);
+void func_800C3E40(AudioMgr* audioMgr);
+void func_800C3E70(AudioMgr* audioMgr);
+void func_800C3FC4(AudioMgr* audioMgr);
+void func_800C3FEC(AudioMgr* audioMgr, void* stack, OSPri pri, OSId id, SchedContext* sched, IrqMgr* irqMgr);
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/pad_800C3C70.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3C80.s")
+//#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3C80.s")
```
Comment (Random06457) [inline src/code/audioMgr.c:14]:
```suggestion
```

File: src/code/audioMgr.c [hunk 2]
```diff
@@ -1,16 +1,130 @@
...
+void func_800C3C80(AudioMgr* audioMgr) {
+    Sub_AudioMgr_18* sub;
+
+    sub = audioMgr->unk_70;
+    if (audioMgr->unk_70->unk_40 != NULL) {
+        osSendMesg(sub->unk_40, NULL, OS_MESG_BLOCK);
+    }
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3CB8.s")
```
Comment (Random06457) [inline src/code/audioMgr.c:24]:
```suggestion
```

File: src/code/audioMgr.c [hunk 3]
```diff
@@ -1,16 +1,130 @@
...
+    D_8016A550 = 0;
+    if (audioMgr->unk_70 != NULL) {
+        osRecvMesg(&audioMgr->unk_AC, NULL, OS_MESG_BLOCK);
+        func_800C3C80(audioMgr);
+    }
+    audioMgr->unk_70 = sub;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3CB8.s")
+//#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3E40.s")
```
Comment (Random06457) [inline src/code/audioMgr.c:58]:
```suggestion
```

File: src/code/audioMgr.c [hunk 4]
```diff
@@ -1,16 +1,130 @@
...
+            continue;
+        }
+        func_800C3E40(audioMgr);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3E70.s")
+#endif
+
+//#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3FC4.s")
```
Comment (Random06457) [inline src/code/audioMgr.c:109]:
```suggestion
```

File: src/code/audioMgr.c [hunk 5]
```diff
@@ -1,16 +1,130 @@
...
+void func_800C3FC4(AudioMgr* audioMgr) {
+    osRecvMesg(&audioMgr->unk_C8, NULL, OS_MESG_BLOCK);
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/func_800C3FEC.s")
```
Comment (Random06457) [inline src/code/audioMgr.c:114]:
```suggestion
```

**Comment by Roman971** [comment]
Move the stuff in sched.h to z64.h, then you can get rid of the audiomgr.h header as well.

File: include/functions.h [hunk 6]
```diff
@@ -1515,12 +1515,12 @@ void THA_Init(TwoHeadArena* tha);
...
-// ? func_800C3E40(?);
-// ? func_800C3FC4(?);
-// ? func_800C3FEC(?);
-// ? func_800C3E70(?);
+// void func_800C3C80(AudioMgr* audioMgr);
+// void AudioMgr_HandleRetrace(AudioMgr* audioMgr);
+// void AudioMgr_HandlePRENMI(AudioMgr* audioMgr);
+// void AudioMgr_ThreadEntry(void* arg0);
+// void AudioMgr_Unlock(AudioMgr* audioMgr);
+// void AudioMgr_Start(AudioMgr* audioMgr, void* stack, OSPri pri, OSId id, SchedContext* sched, IrqMgr* irqMgr);
```
Comment (Roman971) [inline include/functions.h:1523]:
```suggestion
void func_800C3C80(AudioMgr* audioMgr);
void AudioMgr_HandleRetrace(AudioMgr* audioMgr);
void AudioMgr_HandlePRENMI(AudioMgr* audioMgr);
void AudioMgr_ThreadEntry(void* arg0);
void AudioMgr_Unlock(AudioMgr* audioMgr);
void AudioMgr_Start(AudioMgr* audioMgr, void* stack, OSPri pri, OSId id, SchedContext* sched, IrqMgr* irqMgr);
```

File: src/code/audioMgr.c [hunk 7]
```diff
@@ -1,16 +1,116 @@
 #include <ultra64.h>
 #include <global.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audioMgr/pad_800C3C70.s")
+void func_800C3C80(AudioMgr* audioMgr);
+void AudioMgr_HandleRetrace(AudioMgr* audioMgr);
+void AudioMgr_HandlePRENMI(AudioMgr* audioMgr);
+void AudioMgr_ThreadEntry(void* arg0);
+void AudioMgr_Unlock(AudioMgr* audioMgr);
+void AudioMgr_Start(AudioMgr* audioMgr, void* stack, OSPri pri, OSId id, SchedContext* sched, IrqMgr* irqMgr);
```
Comment (Roman971) [inline src/code/audioMgr.c:10]:
```suggestion
```

**Pull request merged**